modules/network/src/main/java/org/apache/ignite/internal/network/direct/stream/DirectByteBufferStreamImplV1.java

```diff
@@ -270,18 +308,18 @@ public void writeBoxedShort(@Nullable Short val) {
     /** {@inheritDoc} */
     @Override
     public void writeInt(int val) {
-        lastFinished = buf.remaining() >= MAX_VAR_INT_BYTES;
+        lastFinished = remainingInternal() >= MAX_VAR_INT_BYTES;
 
-        writeVarInt(Integer.toUnsignedLong(val + 1));
+        writeVarInt(val + 1);
```

Comment:
In all usages of writeVar(Int/Long/Short) we add +1 to val, it always will be the case? Could we do that inside of the methods?